### PR TITLE
IngressVIPs for VMW cluster creation

### PR DESCRIPTION
**Pull request by Randy424**
Regarding: https://issues.redhat.com/browse/ACM-5590

Screenshot 2024-11-07 at 11 13 43 PM

Renders IngressVIPs multitext component as default input for VMW cluster creation, and will revert to single text field IngressVIP when selected OpenShift version is less than 4.12.0, [which is when support for a list of IngressVIP values began](https://docs.openshift.com/container-platform/4.12/installing/installing_vsphere/installing-vsphere-installer-provisioned-customizations.html).

HOW TO TEST:
1. Go to cluster creation
2. Select VMware vSphere
3. Pick an OpenShift version
4. Verify that platform.ingressVIPs is in install-config
5. Move to Network step
6. Fill in multiple ingressVIPs field
7. Verify validation is working
8. Verify Summary page details ingressVIPs information correctly
9. Verify install-config represents ingressVIP yaml list correctly.

Find yaml sample [here](https://docs.openshift.com/container-platform/4.12/installing/installing_vsphere/installing-vsphere-installer-provisioned-customizations.html#installation-installer-provisioned-vsphere-config-yaml_installing-vsphere-installer-provisioned-customizations).

### CODE REVIEW

**Comment by Randy424** [comment]
/hold

Not ready for merge yet. Reviewed with @jeswanke. Added a fix for an existing syntax issue. A reverse function for new component needs to be implemented, and some more refactoring will be necessary to get it working.

File: frontend/src/components/TemplateEditor/TemplateEditor.js [hunk 1]
```diff
@@ -824,14 +825,15 @@ export default class TemplateEditor extends React.Component {
     let editorIndex = this.editors.findIndex((e) => e.id === editor.id)
     if (editorIndex < 0) {
       editorIndex = this.editors.push(editor) - 1
+    } else {
+      this.editors[editorIndex] = editor
     }
-    if (editorIndex > 1) {
-      otherYAMLTabs[editorIndex - 1].editor = editor
+    if (editorIndex >= 1) {
+      set(otherYAMLTabs, `${editorIndex - 1}.editor`, this.editors[editorIndex])
```
Comment (Randy424) [inline frontend/src/components/TemplateEditor/TemplateEditor.js:832]: @jeswanke the defect fix you asked me to include.

**Review by jeswanke — APPROVED** [review]

**Pull request merged**